lbrynet/wallet/resolve.py

```diff
@@ -284,87 +253,56 @@ def prepare_claim_queries(start_position, query_size, channel_claim_infos):
 
 
-def _verify_proof(name, claim_trie_root, result, height, depth, transaction_class, hash160_to_address):
+def _verify_proof(name, claim_trie_root, result, height, depth, transaction_class, ledger):
     """
     Verify proof for name claim
     """
+    support_amount = sum([amt for (stxid, snout, amt) in result['supports']])
 
-    def _build_response(name, value, claim_id, txid, n, amount, effective_amount,
-                        claim_sequence, claim_address, supports):
+    def _build_response(name, tx, nOut):
+        output = tx.outputs[nOut]
         r = {
             'name': name,
-            'value': hexlify(value),
-            'claim_id': claim_id,
-            'txid': txid,
-            'nout': n,
-            'amount': amount,
-            'effective_amount': effective_amount,
+            'value': hexlify(output.script.values['claim']),
+            'claim_id': output.claim_id,
+            'txid': tx.id,
+            'nout': nOut,
+            'amount': output.amount,
+            'effective_amount': output.amount + support_amount,
             'height': height,
             'depth': depth,
-            'claim_sequence': claim_sequence,
-            'address': claim_address,
-            'supports': supports
+            'claim_sequence': result['claim_sequence'],
+            'address': output.get_address(ledger),
+            'supports': result['supports']
         }
         return r
 
     def _parse_proof_result(name, result):
-        support_amount = sum([amt for (stxid, snout, amt) in result['supports']])
-        supports = result['supports']
         if 'txhash' in result['proof'] and 'nOut' in result['proof']:
             if 'transaction' in result:
                 tx = transaction_class(raw=unhexlify(result['transaction']))
                 nOut = result['proof']['nOut']
                 if result['proof']['txhash'] == tx.id:
                     if 0 <= nOut < len(tx.outputs):
-                        claim_output = tx.outputs[nOut]
-                        effective_amount = claim_output.amount + support_amount
-                        claim_address = hash160_to_address(claim_output.script.values['pubkey_hash'])
-                        claim_id = result['claim_id']
-                        claim_sequence = result['claim_sequence']
-                        claim_script = claim_output.script
-                        decoded_name = claim_script.values['claim_name'].decode()
-                        decoded_value = claim_script.values['claim']
-                        if decoded_name == name:
-                            return _build_response(name, decoded_value, claim_id,
-                                                   tx.id, nOut, claim_output.amount,
-                                                   effective_amount, claim_sequence,
-                                                   claim_address, supports)
+                        if tx.outputs[nOut].script.values['claim_name'].decode() == name:
```

Comment:
You can just do `tx.output[nOut].claim_name` there is a `@property` on the Output object for this: 

https://github.com/lbryio/lbry/blob/63568185574afa1d53ac545638c38547177c26d2/lbrynet/wallet/transaction.py#L69

Reply:
great, replacing it